### PR TITLE
#322 Add GeoUnit and OrgUnit Schemas

### PR DESCRIPTION
**Pull request by saurabhere**
Fixes Issue #322 
Introduces new concepts OrgUnit and GeoUnits
Already being used in Profile Schema.
Needs to be included before the Profile schema is stabilized to prevent breaking changes.

To Review: @cdegroot-adobe @trieloff @lrosenthol

### CODE REVIEW

File: schemas/common/geounit.schema.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        },
+        "xdm:label": {
+          "title": "Friendly lable of the geographical unit.",
+          "type": "string",
+          "description": "The user-friendly name for the geographical unit."
+        },
+        "xdm:parentGeoUnit": {
+          "title": "Parent Geographical Unit.",
+          "$ref": "https://ns.adobe.com/xdm/common/geounit",
+          "description": "Parent geographical unit of the current geographical unit in the geographical heirarchy."
```
Comment (trieloff) [inline schemas/common/geounit.schema.json]: Typo: hierarchy.

File: schemas/common/geounit.schema.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/common/geounit",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "type": "object",
+  "meta:extensible": true,
+  "title": "Geo Unit",
+  "description": "The geographical unit of a parent geographical unit.",
```
Comment (trieloff) [inline schemas/common/geounit.schema.json]: Why the reference to parent here? How does this relate to other geo objects? Please elaborate this in the description.

File: schemas/common/orgunit.schema.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/common/orgunit",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "type": "object",
+  "meta:extensible": true,
+  "title": "Org Unit",
```
Comment (trieloff) [inline schemas/common/orgunit.schema.json]: Organizational

File: schemas/common/orgunit.schema.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  "description": "The organisation unit of a parent organisation.",
```
Comment (trieloff) [inline schemas/common/orgunit.schema.json]: Should be "organi*z*ation", US spelling.

File: schemas/context/profile.schema.json [hunk 5]
```diff
@@ -80,17 +80,17 @@
             "$ref": "https://ns.adobe.com/xdm/context/pushnotificationtoken"
           }
         },
-        "xdm:orgUnitID": {
```
Comment (trieloff) [inline schemas/context/profile.schema.json]: This is a breaking change, please update the `CHANGELOG.md`

Comment (saurabhere) [reply schemas/context/profile.schema.json]: OK, breaking changes between releases. Got it.

File: schemas/common/orgunit.schema.json (hunk 4 above)
Comment (saurabhere) [reply schemas/common/orgunit.schema.json]: done!

File: schemas/common/orgunit.schema.json (hunk 3 above)
Comment (saurabhere) [reply schemas/common/orgunit.schema.json]: done!

File: schemas/common/geounit.schema.json (hunk 1 above)
Comment (saurabhere) [reply schemas/common/geounit.schema.json]: :+1:

**Comment by saurabhere** [comment]
fixed build and spellings. @trieloff OK by you?

**Comment by saurabhere** [comment]
@cdegroot-adobe While your suggestion is one way to handle parent-child relationships, what are the downsides to this current approach? It seems clearer to me w.r.t schema in mind.

File: schemas/context/profile.schema.json (hunk 5 above)
Comment (trieloff) [reply schemas/context/profile.schema.json]: Please see fa2c6f4 for how to document the breaking part of the change.

**Comment by trieloff** [comment]
@saurabhere approved. Please resolve the conflict.

@cdegroot-adobe I see that @saurabhere's approach to modeling the relationship is in line with @kstreeter's suggestions, so I don't think there is an issue.

File: schemas/context/profile.schema.json (hunk 5 above)
Comment (saurabhere) [reply schemas/context/profile.schema.json]: thanks. keeping in mind for future.

**Comment by trieloff** [comment]
@saurabhere please check this https://github.com/adobe/xdm/pull/307#discussion_r191040082 with @anshuman-yadav – it looks like you will need to update the `extensions/adobe/experience/campaign/orchestration/reportingevent.schema.json`, too.

Once this is done, we can merge, i.e. I'm dismissing @cdegroot-adobe's requested changes.

**Pull request merged**